pkg/predictor_source/inferenceservice_registry.go

```diff
@@ -68,6 +69,55 @@ func BuildBasePredictorFromInferenceService(isvc *v1beta1.InferenceService) (*v1
 	return p, nil
 }
 
+// Return secretKey, bucket, modelPath, schemaPath, and error
+func processInferenceServiceStorage(inferenceService *v1beta1.InferenceService, nname types.NamespacedName) (string, string, string, string, error) {
+	var bucket string
+	var modelPath string
+	var schemaPath string
+	secretKey := inferenceService.ObjectMeta.Annotations[v1beta1.SecretKeyAnnotation]
+	_, frameworkSpec := inferenceService.Spec.Predictor.GetPredictorFramework()
+	storageUri := frameworkSpec.StorageURI
+	storageSpec := frameworkSpec.Storage
+	if storageUri == nil {
+		if storageSpec != nil && storageSpec.Path != nil {
+			modelPath = *storageSpec.Path
+			if storageSpec.SchemaPath != nil {
+				schemaPath = *storageSpec.SchemaPath
+			}
+		} else {
+			return "", "", "", "", fmt.Errorf("the InferenceService %v must have either the storageUri or the storage.path", nname)
+		}
```

Comment:
I still think it would be cleaner to invert this one, then you can eliminate an `else` block. Any reason for not doing that?